lib/appcanary/configuration.rb

```diff
@@ -0,0 +1,51 @@
+require "yaml"
+
+module Appcanary
+  APPCANARY_DEFAULT_BASE_URI = "https://appcanary.com/api/v3/monitors"
+
+  class Configuration
+    attr_accessor :base_uri, :api_token, :monitor_name
+
+    def initialize
+      self.base_uri = APPCANARY_DEFAULT_BASE_URI
+    end
+  end
+
+  class ConfigurationError < RuntimeError
+  end
+
+  class << self
```

Comment:
(Ruby syntax pedantry: the important part here is it's modifying the "scope" of the method being defined. Fun pro tip: you can define a method on _any_ object, i.e. 

```ruby
[1] pry(main)> an_object = "lol"
=> "lol"
[2] pry(main)> def an_object.foo; puts "notlol"; end
=> :foo
[3] pry(main)> an_object.foo
notlol
=> nil
[4] pry(main)> "lol".foo
NoMethodError: undefined method `foo' for "lol":String
from (pry):4:in `__pry__'
```

so , in `def self.foo` self refers to the class instance you're currently scoped in.